asserts/assertstest/assertstest.go

```diff
@@ -141,3 +143,185 @@ func GPGImportKey(homedir, armoredKey string) {
 		panic(fmt.Errorf("cannot import test key into GPG setup at %q: %v (%q)", homedir, err, out))
 	}
 }
+
+// A SignerDB can sign assertions using its key pairs.
+type SignerDB interface {
+	Sign(assertType *asserts.AssertionType, headers map[string]string, body []byte, keyID string) (asserts.Assertion, error)
+}
+
+// NewAccount creates an account assertion for username, it fills in values for other missing headers as needed. It panics on error.
+func NewAccount(db SignerDB, username string, headers map[string]string, keyID string) *asserts.Account {
+	if headers == nil {
+		headers = make(map[string]string)
+	}
+	headers["username"] = username
+	if headers["account-id"] == "" {
+		headers["account-id"] = strutil.MakeRandomString(32)
+	}
+	if headers["display-name"] == "" {
+		headers["display-name"] = strings.ToTitle(username)
+	}
+	if headers["validation"] == "" {
+		headers["validation"] = "unproven"
+	}
+	if headers["timestamp"] == "" {
+		headers["timestamp"] = time.Now().Format(time.RFC3339)
+	}
+	a, err := db.Sign(asserts.AccountType, headers, nil, keyID)
+	if err != nil {
+		panic(err)
+	}
+	return a.(*asserts.Account)
+}
+
+// NewAccountKey creates an account-key assertion for the account, it fills in values for missing headers as needed. In panics on error.
+func NewAccountKey(db SignerDB, acct *asserts.Account, pubKey asserts.PublicKey, headers map[string]string, keyID string) *asserts.AccountKey {
```

Comment:
well you need an account assertion to use an account key so needing acct forces that as well which is good